### PR TITLE
Implement contact attribute and skip on/off configuration for Philips SOC001

### LINKED ISSUE #3314: [Device Support Request] SOC001 - Philips Contact sensor
### Problem description

Signify released a while a go a contact sensor to extend their security product portfolio.
Unfortunately they almost exclusively putted all relevant attributes in their manufacturer cluster.
Good news is that the device seems to overall have a good behaviour continuing the reputation of Signify.

### Solution description

1. Implement the manufacturer specific cluster :
- PhilipsContactCluster : 0xFC06

2. Implements the associated attributes :
- Contact
Attribute: 0x0100
Data Type: 8-Bit Enumeration (0x30)
Uint8: 1 (0x01)

- Last contact change (10th of a second)
Attribute: 0x0101
Data Type: 32-Bit Unsigned Integer (0x23)
Uint32: 0 (0x00000000)

- Tamper
Attribute: 0x0102
Data Type: 8-Bit Enumeration (0x30)
Uint8: 0 (0x00)

- Last tamper change (10th of a second)
Attribute: 0x0103
Data Type: 32-Bit Unsigned Integer (0x23)
Uint32: 12425 (0x00003089)

3. Handle the on_off cluster as it is stateless and only meant for biding with lights.

### Screenshots/Video

<details><summary>Screenshots/Video</summary>

[Paste/upload your media here]

</details>

### Device signature

<details><summary>Device signature</summary>

```json
{
  "node_descriptor": {
    "logical_type": 2,
    "complex_descriptor_available": 0,
    "user_descriptor_available": 0,
    "reserved": 0,
    "aps_flags": 0,
    "frequency_band": 8,
    "mac_capability_flags": 128,
    "manufacturer_code": 4107,
    "maximum_buffer_size": 82,
    "maximum_incoming_transfer_size": 128,
    "server_mask": 11264,
    "maximum_outgoing_transfer_size": 128,
    "descriptor_capability_field": 0
  },
  "endpoints": {
    "2": {
      "profile_id": "0x0104",
      "device_type": "0x0402",
      "input_clusters": [
        "0x0000",
        "0x0001",
        "0x0003",
        "0xfc06"
      ],
      "output_clusters": [
        "0x0000",
        "0x0003",
        "0x0006",
        "0x0019"
      ]
    }
  },
  "manufacturer": "Signify Netherlands B.V.",
  "model": "SOC001",
  "class": "soc001.SignifyContact"
}
```

</details>

### Diagnostic information

<details><summary>Diagnostic information</summary>

```json
None
```

</details>

### Logs

<details><summary>Logs</summary>

```python
None
```

</details>

### Custom quirk

<details><summary>Custom quirk</summary>

```python
"""Signify SOC001 device."""

from zigpy.quirks.v2 import QuirkBuilder,ClusterType,BinarySensorDeviceClass
import zigpy.types as types
from zigpy.quirks import CustomCluster

class PhilipsManufacturerSpecific(CustomCluster):
    """Philips Contact manufacturer cluster."""

    cluster_id: types.uint16_t = 0xFC06
    name = "Philips Manufacturer specific"
    ep_attribute = "philips_manufacturer_specific"
    attributes = {
        0x0100: ("contact", types.enum8, True),
        0x0101: ("last_contact_change", types.uint32_t, True),
        0x0102: ("tamper", types.enum8, True),
        0x0103: ("last_tamper_change", types.uint32_t, True),
    }

(
    QuirkBuilder("Signify Netherlands B.V.", "SOC001")
    .removes(
        cluster_id=6,
        endpoint_id=2,
        cluster_type=ClusterType.Client
    )
    .replaces(
        replacement_cluster_class=PhilipsManufacturerSpecific,
        cluster_id=64518,
        endpoint_id=2
    )
    .binary_sensor(
        attribute_name="contact",
        cluster_id=64518, #0xfc06,
        endpoint_id=2,
        device_class=BinarySensorDeviceClass.OPENING
    )
    .binary_sensor(
        attribute_name="tamper",
        cluster_id=64518, #0xfc06,
        endpoint_id=2,
        device_class=BinarySensorDeviceClass.TAMPER
    )
    .add_to_registry()
)
```

</details>

### Additional information

Unfortunately the device does not respond it's attributes whatsoever, probably to make our life a little bit harder.
To discover the attributes, I iterated all attribute addresses and read them to get the following data :

-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
2024-08-19T04 | 37 | 20.475616+00 | 0 | Manufacturer Specific | 256 | enum8.undefined_0x00 | 0x0100 | 0xFC06 | 2 | 0 | 17 | 88 | 1 | 0d | ce | f9 | 51 | 0x100B | 0x30
2024-08-19T04 | 37 | 25.483887+00 | 0 | Manufacturer Specific | 257 | 47103 | 0x0101 | 0xFC06 | 2 | 0 | 17 | 88 | 1 | 0d | ce | f9 | 51 | 0x100B | 0x23
2024-08-19T04 | 37 | 30.487990+00 | 0 | Manufacturer Specific | 258 | enum8.undefined_0x00 | 0x0102 | 0xFC06 | 2 | 0 | 17 | 88 | 1 | 0d | ce | f9 | 51 | 0x100B | 0x30
2024-08-19T04 | 37 | 35.495690+00 | 0 | Manufacturer Specific | 259 | 172317 | 0x0103 | 0xFC06 | 2 | 0 | 17 | 88 | 1 | 0d | ce | f9 | 51 | 0x100B | 0x23

But realised afterward that ebaauw did all the work here : https://github.com/dresden-elektronik/deconz-rest-plugin/issues/7226

### PR DESCRIPTION
**Pull request by mguaylam**
## Proposed change

Implement manufacturer specific contact attribute and skip on/off configuration for SOC001.
Since the on/off attribute does not support reporting configuration, network failure or shutdown will leave Home Assistant unaware of the new sensor state. Demonstrated here : https://github.com/zigpy/zha-device-handlers/pull/3432#issuecomment-2423011662

## Additional information

Confirm binding works on the custom on/off. Tested between a light and this sensor.
Breaking change for previous installations by disabling the on/off entity creation.
Fixes https://github.com/zigpy/zha-device-handlers/issues/3314

## Checklist

- [X] The changes are tested and work correctly
- [X] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

### CODE REVIEW

**Comment by mguaylam** [comment]
Should this go ahead because of https://github.com/zigpy/zha/issues/289 ?

File: zhaquirks/philips/soc001.py [hunk 1]
```diff
@@ -44,14 +61,26 @@ class AttributeDefs(BaseAttributeDefs):
...
+        cluster_type=ClusterType.Client,
+        endpoint_id=2,
+    )
     .replaces(PhilipsContactCluster, endpoint_id=2)
+    .binary_sensor(
+        "contact",
+        PhilipsContactCluster.cluster_id,
+        endpoint_id=2,
+        device_class=BinarySensorDeviceClass.OPENING,
+        EntityType=EntityType.STANDARD,
```
Comment (puddly) [inline zhaquirks/philips/soc001.py]:
```suggestion
        entity_type=EntityType.STANDARD,
```

**Comment by TheJulianJES** [comment]
We've discussed an alternative approach regarding this. Marking this PR as draft.

File: zhaquirks/philips/soc001.py (hunk 1 above)
Comment (mguaylam) [reply zhaquirks/philips/soc001.py]: Thank you.

**Pull request closed without merging**